app/templates/components/request/config-form.hbs

```diff
@@ -0,0 +1,144 @@
+<TravisForm
+  @onSubmit={{perform this.triggerBuild}}
+  as |form|
+>
+  <div class="request-configs-fields" data-test-build-config-form='true'>
+    <div class="form-elem">
+      {{#radio-button
+        value="sha"
+        groupValue=refType
+        changed=(action "change" "refType")
+        classNames="ref-type"
+      }}
+        Commit
+      {{/radio-button}}
```

Comment:
If possible, it's preferred to use angle bracket syntax for components. In this case, it would be something like:
```
<RadioButton
  @value='sha'
  @groupValue={{refType}}
  @changed={{action 'change' refType}}
  @classNames='ref-type'
>
  Commit
</RadioButton>
```

Additionally, in this case I would lean towards making our own component rather than using `ember-radio-button`. We have similar components in `app/components/forms` that we could base a radio component on, and I have some concerns about how actively maintained the `ember-radio-button` addon is. But also, I know that making such a component can be some complex work sometimes.

Reply:
i understand the concerns about adding a dependency that isn't maintained actively. i'll try to remove it.

out of curiosity, what's the reasoning behind preferring the angle bracket syntax? my impression was that most documentation used the curly braces (helper?) syntax.

Reply:
Angle bracket syntax is a somewhat recent addition to ember, so older addons, forum posts etc. will tend to have the curly brace syntax. It seems to be the way the ember community is going now, and IIRC one of the main ideas was to make it easier to differentiate components from other handlebars things (if/else, helpers, etc). It also allows clearer separation between parameters and html attributes. Here's some more info:

https://guides.emberjs.com/v3.17.0/upgrading/current-edition/templates/#toc_benefits-of-angle-brackets

Reply:
i actually like this much better, too. thanks for the pointer, i'll have a look!

Reply:
Awesome :smile: I feel like I should mention, the guide I linked is for an ember version that is a few ahead of where we're at (3.12). But I think everything on that particular page is applicable to 3.12, and it has a great summary of the advantages of angle bracket syntax

Reply:
addressed in https://github.com/travis-ci/travis-web/commit/4a7aa854a101dc651328fe30943edbb727ddccd4

Reply:
@svenfuchs how do you feel about trying to use our existing [form-checkbox](https://github.com/travis-ci/travis-web/blob/master/app/components/forms/form-checkbox.js) component instead of that outdated add-on? It looks very much like a radio (except it has a checkmark, but that shouldn't be an issue) and it's easy to make it behave like a radio (uncheck others on check).